blueman/gui/manager/ManagerMenu.py

```diff
@@ -18,88 +18,115 @@ class ManagerMenu:
     def __init__(self, blueman):
         self.blueman = blueman
 
-        self.Menubar = blueman.Builder.get_object("menu")
-
-        self.adapter_items = []
+        self.adapter_items = {}
+        self._adapters_group = []
+        self._insert_adapter_item_pos = 2
         self.Search = None
 
-        self.item_adapter = Gtk.MenuItem.new_with_mnemonic(_("_Adapter"))
-        self.item_device = Gtk.MenuItem.new_with_mnemonic(_("_Device"))
-
-        self.item_view = Gtk.MenuItem.new_with_mnemonic(_("_View"))
-        self.item_help = Gtk.MenuItem.new_with_mnemonic(_("_Help"))
+        self.item_adapter = self.blueman.Builder.get_object("item_adapter")
+        self.item_device = self.blueman.Builder.get_object("item_device")
 
-        self.Menubar.append(self.item_adapter)
-        self.Menubar.append(self.item_device)
-        self.Menubar.append(self.item_view)
-        self.Menubar.append(self.item_help)
+        self.item_view = self.blueman.Builder.get_object("item_view")
+        self.item_help = self.blueman.Builder.get_object("item_help")
 
         help_menu = Gtk.Menu()
 
         self.item_help.set_submenu(help_menu)
         help_menu.show()
 
-        item = create_menuitem(_("_Report a Problem"), get_icon("dialog-warning", 16))
-        item.connect("activate", lambda x: launch("xdg-open %s/issues" % WEBSITE, None, True))
-        help_menu.append(item)
-        item.show()
+        report_item = create_menuitem(_("_Report a Problem"), get_icon("dialog-warning", 16))
+        report_item.show()
+        help_menu.append(report_item)
```

Comment:
Theoretically you display a visible item in an already visible menu here that does not react to its activation yet. Just mentioning, not relevant as nobody will click that fast. ;)